delta/sourcing-psql/src/main/scala/ch/epfl/bluebrain/nexus/delta/sourcing/model/Envelope.scala

```diff
@@ -19,4 +30,43 @@ import java.time.Instant
   * @param offset
   *   the offset
   */
-final case class Envelope[Id, Value](tpe: EntityType, id: Id, rev: Int, value: Value, instant: Instant, offset: Offset)
+final case class Envelope[Id, +Value](tpe: EntityType, id: Id, rev: Int, value: Value, instant: Instant, offset: Offset)
+
+object Envelope {
+
+  def stream[Id, Value](offset: Offset, query: Offset => Fragment, strategy: RefreshStrategy, xas: Transactors)(implicit
```

Comment:
No sure if this is better defined here (compared to previous).

Reply:
This thing will be used by every store to stream events/states so I chose to isolate this logic into a function that takes an offset and can create a query from an offset.